### PR TITLE
Fix irisws-distaz client following service upgrade to 1.0.3

### PR DESCRIPTION
**Pull request by chad-iris**
The previous irisws-distaz client used an undocumented feature to request JSON results. After a recent update to the service this feature is no longer supported.

This PR changes the code to parse the default XML. Also, 'distancemeters' and 'ellipsoidname' are now returned in the result. Tests have been updated.

### CODE REVIEW

File: obspy/clients/iris/client.py [hunk 1]
```diff
@@ -558,7 +558,7 @@ def sacpz(self, network, station, location="*", channel="*",
     def distaz(self, stalat, stalon, evtlat, evtlon):
         """
         Low-level interface for `distaz` Web service of IRIS
-        (http://service.iris.edu/irisws/distaz/) - release 1.0.1 (2010).
+        (http://service.iris.edu/irisws/distaz/) - release 1.0.3 (2016).
```
Comment (QuLogic) [inline obspy/clients/iris/client.py]: Since you're changing this line, we should use the https link.

File: obspy/clients/iris/client.py [hunk 2]
```diff
@@ -583,34 +583,41 @@ def distaz(self, stalat, stalon, evtlat, evtlon):
         Latitudes are converted to geocentric latitudes using the WGS84
         spheroid to correct for ellipticity.
 
+        The distance (in degrees) and distancemeters are the distance
```
Comment (QuLogic) [inline obspy/clients/iris/client.py]: Can you put the two names in quotes or double backticks, so it's clear that `distancemeters` is not a typo?

File: obspy/clients/iris/tests/test_client.py [hunk 3]
```diff
@@ -58,14 +58,18 @@ def test_distaz(self):
         client = Client()
         # normal request
         result = client.distaz(stalat=1.1, stalon=1.2, evtlat=3.2, evtlon=1.4)
-        self.assertAlmostEqual(result['distance'], 2.09554)
-        self.assertAlmostEqual(result['backazimuth'], 5.46946)
-        self.assertAlmostEqual(result['azimuth'], 185.47692)
+        self.assertAlmostEqual(result['distance'], 2.10256)
```
Comment (QuLogic) [inline obspy/clients/iris/tests/test_client.py]: Is there a reason for the change in results? Just more correct? The new one seems to match geographiclib at least.

File: obspy/clients/iris/client.py (hunk 1 above)
Comment (chad-iris) [inline obspy/clients/iris/client.py]: HTTPS is experimental and not a fully tested or supported for IRIS services.

File: obspy/clients/iris/tests/test_client.py (hunk 3 above)
Comment (chad-iris) [inline obspy/clients/iris/tests/test_client.py]: The service now uses GeographicLib and is (presumably) more accurate.

File: obspy/clients/iris/client.py (hunk 2 above)
Comment (chad-iris) [inline obspy/clients/iris/client.py]: Done.  Also for the references to the other keys in the result for consistency.

**Comment by krischer** [comment]
I added one more commit as the doctest for the traveltime service somehow required a slightly different rounding. I guess this is due to some service change? Could have been a while ago as we don't regularly run the networks tests with our CI runners as they just tend to be too unstable.

Side note: should we maybe have some test runners that run all the network tests at least once a day?

**Comment by krischer** [comment]
Thanks for this!

**Comment by megies** [comment]
> Side note: should we maybe have some test runners that run all the network tests at least once a day?

We could do this with our docker testbot that currently runs once a day..

**Comment by krischer** [comment]
Would be great :-) Then we can at least manually check if some stop working completely.

**Pull request merged**